feat: add artist radio and track radio (in context menu)

farfromrefug opened the pull request:
This PR adds:

* a new Artist Radio button in Artist page header
* a new context menu on tracks to start track radio

The artist radio use appropriate API for all server kind.
I am just not sure about the artist radio button style. Could not get it to use the same style as play button. But do we want the same style?

Possible improvements:
* add artist radio entry in the context menu for artists
* add artist radio button to artist list item "overlay" (with play button and others)

Copilot committed:
Initial plan

Copilot committed:
Restore deb package building for Linux with auto-update support

Co-authored-by: farfromrefug <655344+farfromrefug@users.noreply.github.com>

farfromrefug committed:
Merge pull request #1 from farfromrefug/copilot/restore-deb-package-building

Restore deb package building for Linux

Copilot committed:
Initial plan

Copilot committed:
Add API support for artist radio and track radio features

Co-authored-by: farfromrefug <655344+farfromrefug@users.noreply.github.com>

Copilot committed:
Add translation strings and settings UI for artist radio count

Co-authored-by: farfromrefug <655344+farfromrefug@users.noreply.github.com>

Copilot committed:
Address code review feedback: remove redundant check and use radio icon

Co-authored-by: farfromrefug <655344+farfromrefug@users.noreply.github.com>

farfromrefug committed:
Merge branch 'jeffvli:development' into development

farfromrefug committed:
Merge remote-tracking branch 'origin/development' into copilot/add-artist-radio-feature

# Conflicts:
#	src/renderer/api/query-keys.ts
#	src/renderer/features/artists/components/album-artist-detail-content.tsx
#	src/renderer/features/settings/components/general/control-settings.tsx
#	src/renderer/features/songs/api/songs-api.ts

farfromrefug committed:
chore: lint fixes

jeffvli commented on src/renderer/features/artists/components/album-artist-detail-header.tsx:
```diff
@@ -63,8 +65,31 @@ export const AlbumArtistDetailHeader = forwardRef((_props, ref: Ref<HTMLDivEleme
+                    },
+                    serverId: server.id,
+                }),
+            );
```
I would like to see the `queryKeys.player.fetch` query key utilized on these fetches, as well as attaching loading state to the button using the `useIsPlayerFetching` hook, otherwise the user can press the button multiple times without any indication that a fetch is happening.

farfromrefug replied:
i added the missing loading indicator

jeffvli commented on src/renderer/features/context-menu/actions/play-track-radio-action.tsx:
```diff
@@ -0,0 +1,51 @@
+                    },
+                    serverId,
+                }),
+            );
```
Same as the other comment - attach the `queryKeys.player.fetch` query key to this fetch.

jeffvli commented on src/renderer/features/context-menu/menus/song-context-menu.tsx:
```diff
@@ -28,6 +29,7 @@ export const SongContextMenu = ({ items, type }: SongContextMenuProps) => {
             bottomStickyContent={<ContextMenuPreview items={items} itemType={type} />}
         >
             <PlayAction ids={ids} itemType={LibraryItem.SONG} songs={items} />
+            <PlayTrackRadioAction song={items[0]} />
```
It would also make sense to add this to the QueueContextMenu and PlaylistSongContext menus as those are both derivatives of the `Song` type.

farfromrefug replied:
@jeffvli sure but QeueContextMenu and PlaylistSongContext are for multiple ids? How would i choose the one to pass to PlayTrackRadioAction?

jeffvli replied:
The same way it's being handled in the SongContextMenu? Either pick the first item, or disable the item if more than one is selected. Preferably disable now that I think about it.

farfromrefug replied:
@jeffvli done!

jeffvli commented on src/renderer/features/shared/components/library-header.tsx:
```diff
@@ -234,6 +239,21 @@ export const LibraryHeaderMenu = ({
                 {onPlay && (
                     <PlayLastTextButton {...handlePlayLast.handlers} {...handlePlayLast.props} />
                 )}
+                {onArtistRadio && (
+                    <Button
+                        className={clsx(styles.wideTextButton, styles.unthemed)}
+                        classNames={{
+                            label: styles.wideTextButtonLabel,
+                            root: styles.wideTextButton,
+                        }}
+                        leftSection={<Icon icon="radio" />}
+                        onClick={onArtistRadio}
+                        size="md"
+                        variant="subtle"
+                    >
+                        {t('page.albumArtistDetail.artistRadio', { postProcess: 'sentenceCase' })}
+                    </Button>
+                )}
```
I'm not sold on the design here.

I would:
1. remove the className and classNames as they are specifically only applied for the special text play button styles.
2. Use variant `transparent` instead

farfromrefug committed:
chore: changes per comments

farfromrefug committed:
chore: address comments

farfromrefug commented:
@jeffvli i added a new menu entry for artist context menu to start radio.

jeffvli commented on src/renderer/features/context-menu/menus/song-context-menu.tsx:
```diff
@@ -28,6 +29,7 @@ export const SongContextMenu = ({ items, type }: SongContextMenuProps) => {
             bottomStickyContent={<ContextMenuPreview items={items} itemType={type} />}
         >
             <PlayAction ids={ids} itemType={LibraryItem.SONG} songs={items} />
+            <PlayTrackRadioAction song={items[0]} />
```
This also needs to be disabled to match the others.

jeffvli commented on src/renderer/features/context-menu/menus/artist-context-menu.tsx:
```diff
@@ -30,6 +31,7 @@ export const ArtistContextMenu = ({ items, type }: ArtistContextMenuProps) => {
             <PlayAction ids={ids} itemType={LibraryItem.ARTIST} />
             <ContextMenu.Divider />
             <AddToPlaylistAction items={ids} itemType={LibraryItem.ARTIST} />
+            <PlayArtistRadioAction artist={items[0]} disabled={items.length > 1} />
```
Also add this to the album-artist-context-menu. Most of the app uses album artist as opposed to artist.

jeffvli commented:
Another aspect of this feature that I feel may be desirable is to be able to add the additional play options (play / next / last) when starting the radio, in case the user does not want to overwrite their entire queue.

If you don't feel comfortable implementing this, let me know. It's not a requirement, but definitely a nice to have.

farfromrefug committed:
chore: changes per comments

farfromrefug committed:
core: more context menu actions

farfromrefug committed:
chore: loading on artist radio button

farfromrefug commented:
@jeffvli i addressed all comments. As per additional play options (play / next / last) to be honest i am not sure what you are thinking about. So i would prefer to leave it to you if that s ok.
One thing i was thinking about is an option to open fullscreen player when started. Right now you might be wondering what s happening

jeffvli committed:
Merge branch 'development' into copilot/add-artist-radio-feature

jeffvli committed:
add play types for artist/track radio
